crawler/curriculum/curriculum.go

```diff
@@ -0,0 +1,475 @@
+	columns.Each(func(columnIndex int, column *goquery.Selection) {
+		if _, ok := columnMap[columnIndex]; ok {
+			var element *goquery.Selection
+
+			if column.Find("a").Length() == 0 {
+				element = column
+			} else {
+				element = column.Find("a")
+			}
+
+			if columnIndex >= 8 && columnIndex <= 14 {
+				handlePeriods(columnIndex, element, hasSundayCourses, hasSaturdayCourses, course)
+			} else if columnIndex == 6 || columnIndex == 15 {
+				handleInstructorAndClassroom(element, columnIndex, course)
+			} else {
+				handleIDAndName(element, columnIndex, course)
+			}
+		}
+	})
+}
+
+func handlePeriods(
+	columnIndex int,
+	element *goquery.Selection,
+	hasSundayCourses *bool,
+	hasSaturdayCourses *bool,
+	course *Course,
+) {
+	day := columnIndex - 8
+	big5Element, _ := decoder.DecodeToBig5(element.Text())
+	course.Periods[day] = strings.TrimSpace(big5Element)
+
+	if !(*hasSaturdayCourses) && day == 6 && len(course.Periods[day]) != 0 {
+		*hasSaturdayCourses = true
+	}
+
+	if !(*hasSundayCourses) && day == 0 && len(course.Periods[day]) != 0 {
+		*hasSundayCourses = true
+	}
+}
+
+func handleInstructorAndClassroom(
+	element *goquery.Selection,
+	columnIndex int,
+	course *Course,
+) {
+	element.Each(func(i int, el *goquery.Selection) {
+		switch columnMap[columnIndex] {
+		case "instructor":
+			big5Instructor, err := decoder.DecodeToBig5(el.Text())
+			if err != nil {
+				log.Panicln(err)
+				break
+			}
+			course.Instructor = append(course.Instructor, strings.TrimSpace(big5Instructor))
+		case "classroom":
+			big5Classroom, err := decoder.DecodeToBig5(el.Text())
+			if err != nil {
+				log.Panicln(err)
+				break
+			}
+			course.Classroom = append(course.Classroom, strings.TrimSpace(big5Classroom))
+		default:
+			log.Println("beyond the map", columnMap[columnIndex])
+		}
+	})
+}
+
+func handleIDAndName(
+	element *goquery.Selection,
+	columnIndex int,
+	course *Course,
+) {
+	switch columnMap[columnIndex] {
+	case "id":
+		big5Id, err := decoder.DecodeToBig5(element.Text())
+		if err != nil {
+			log.Panicln(err)
+			break
+		}
+		course.ID = strings.TrimSpace(big5Id)
+	case "name":
+		big5Name, err := decoder.DecodeToBig5(element.Text())
+		if err != nil {
+			log.Panicln(err)
+			break
+		}
+		course.Name = strings.TrimSpace(big5Name)
+	default:
+		log.Println("beyond the map", columnMap[columnIndex])
+	}
+}
+
+func handleCourseSelectRequest(
+	studentID string,
+	year string,
+	sem string,
+) (buffer *http.Response, err error) {
+	bufferReq, err := http.NewRequest("GET", config.CoureseSystem.Select, nil)
+	if err != nil {
+		log.Panicln(err)
+		return nil, err
+	}
+
+	bufferReq.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	q := bufferReq.URL.Query()
+	q.Add("format", "-2")
+	q.Add("code", studentID)
+	q.Add("year", year)
+	q.Add("sem", sem)
+
+	bufferReq.URL.RawQuery = q.Encode()
+
+	buffer, err = client.Do(bufferReq)
+	if err != nil {
+		log.Panicln(err)
+		return nil, err
+	}
+
+	return buffer, nil
+}
```

Comment:
Will it be better to separate the curriculum and curriculum course to different files?  I think the file now is too large to read and maintain.

Reply:
I've thought about this issue, but I think I will resolve it just after I finish another api.

Reply:
OK, then I will first skip this file because it is really hard to review XD

Reply:
you could just clone it or open it on `Gitpod`, make sure that the logic is work fine or is there something that I just miss.